File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
class Solution {
    public long solution(int balls, int share) {
        long n = Math.min(share, balls - share);
        long temp = n;
        long up = 1;
        long bot = 1;
        for(long i = n; i >= 1; i--){
            up *= balls--;
            bot *= temp--;
            if(up % bot == 0) {
                up /= bot;
                bot /= bot;
            }
        }
        up/=bot;
        return up;
    }
}
/*
nCm
1, 2, 3, 4, 5
1, 2, 

*/

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.py

def solution(balls, share):
    r = min(share, balls - share)
    n = balls
    top, bottom = 1, 1
    while r > 0:
        top *= n
        bottom *= r
        if top % bottom == 0: 
            top // bottom
        n -= 1
        r -= 1
    return top // bottom